### ISSUE TITLE
Toolbar buttons do not render nicely

### ISSUE THREAD

**Issue by Chrriis** [JFormDesigner/FlatLaf#56]
Toolbar buttons feel a bit small and toggle buttons do not have a nice look. Defaults should be more pleasing to the eye.

Here is the current toolbar:
![FlatLafToolBarButtons2](https://user-images.githubusercontent.com/1174202/73726761-63683f00-4730-11ea-884e-ec8c04c775cf.png)

And here is the corresponding test case:
```Java
		UIManager.setLookAndFeel(new FlatLightLaf());
		JFrame frame = new JFrame();
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		JPanel panel = new JPanel();
		JToolBar toolBar = new JToolBar();
		JToggleButton button1 = new JToggleButton(new FlatFileViewFloppyDriveIcon());
		button1.setSelected(true);
		toolBar.add(button1);
		JToggleButton button2 = new JToggleButton(new FlatFileViewComputerIcon());
		button2.setSelected(true);
		toolBar.add(button2);
		JToggleButton button3 = new JToggleButton(new FlatFileViewDirectoryIcon());
		button3.setSelected(true);
		toolBar.add(button3);
		JToggleButton button4 = new JToggleButton(new FlatFileViewHardDriveIcon());
		toolBar.add(button4);
		panel.add(toolBar);
		frame.getContentPane().add(panel);
		frame.setSize(400, 300);
		frame.setVisible(true);
```

Here is a difference between FlatLaf and Windows L&F:
![FlatLafToolBarButtons](https://user-images.githubusercontent.com/1174202/73726534-07051f80-4730-11ea-93f7-7c7fcf04ab72.png)

I think a simple improvement would be for toolbar buttons to have a small empty border around them.

**Comment by smileatom** [comment]
I personally would prefer this be a simple option of supplying a border to toolbar button.  WindowsL&F is not a flat look and feel.

**Comment by Chrriis** [comment]
@smileatom The problem is not Windows vs Flat. If you look at the toolbar at the very top, we have a compact gray smudge with toggle buttons. Generally in flat design, things are flat but not compact. This lacks spacing. As for the Windows comparison, I was merely pointing out that its usability is much better.

**Comment by smileatom** [comment]
I hear what you are saying but the Mac L&F has the border your talking about and it looks pretty bad.

**Comment by Chrriis** [comment]
@smileatom I suggested either a border or empty border. I think empty border would go better with the idea of "flat".

Here are the same screenshots without and with empty borders:
![FlatLafToolBarButtons2_b](https://user-images.githubusercontent.com/1174202/73884901-176be600-4867-11ea-9442-56e155f72ffd.png)
![FlatLafToolBarButtons_b](https://user-images.githubusercontent.com/1174202/73884923-205cb780-4867-11ea-8b9b-67360003f77f.png)

I think buttons are easier to identify, aim at, and it looks less like a gray caterpillar 🙂

**Comment by smileatom** [comment]
@Chrriis Im probably not seeing this with the borders applied by JFormDesigner, Ill test some with no borders and see the result your seeing.  I agree that a visually satisfying border should probably be the default.

**Comment by DevCharly** [comment]
Thanks for the feedback.

I've added some empty space around each toolbar button (1,2,1,2) and also around the whole toolbar (2,2,2,2). This is the same spacing as used in IDEA.

![image](https://user-images.githubusercontent.com/5604048/74536567-a5676100-4f38-11ea-8bc0-d44dfc8a54f7.png)

This looks much better 👍 

There is also a `EmptyBorder` in the UI defaults (key `ToolBar.spacingBorder`) that can be used for custom components in the toolbar to get same spacing between button and custom component (see screenshot above).

Here is an example how to use `ToolBar.spacingBorder` (an additional panel is required):
~~~java
JComboBox<String> comboBox = new JComboBox<>();
comboBox.setModel( new DefaultComboBoxModel<>( new String[] { "hello" } ) );

JPanel panel = new JPanel( new BorderLayout() );
panel.setBorder( UIManager.getBorder( "ToolBar.spacingBorder" ) );
panel.add( comboBox, BorderLayout.CENTER );

toolBar.add( panel );
~~~

Note that if your application supports LaF switching (without restart), the spacing border is lost because it is an instance of `UIResource`. This is good when switching to an non-FlatLaf LaF. Anyway to make it LaF switching safe you have to set the border in method `updateUI()`. E.g.

~~~java
JPanel panel = new JPanel( new BorderLayout() ) {
    @Override
    public void updateUI() {
        super.updateUI();
        setBorder( UIManager.getBorder( "ToolBar.spacingBorder" ) );
    }
};
~~~

**Comment by Chrriis** [comment]
Hi @DevCharly, I tried your changes and the toolbar renders much more nicely!

If I wanted to be picky, I would say that I expected an empty gap around, and not an empty border. The distinction is that a gap does not activate the button on hover, while the empty border does:
![ToolBarEmptyBorder](https://user-images.githubusercontent.com/1174202/74555836-e5403f80-4f5c-11ea-9931-719a90f7bab8.png)
... but I can live with that, this is minor 🙂

**Comment by DevCharly** [comment]
Adding gaps was also my first idea and I already had an layout manager (mostly) implemented, but I had concerns that this is different to all other LaFs and may cause problems when people put custom components into toolbars and do not want a gap between two custom components.

So I tried the IntelliJ way, which works good.
But adding custom components is not that optimal...

Maybe have to rethink the implementation...

**Comment by DevCharly** [comment]
released in 0.27
